docs/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/graphql.md

```diff
@@ -0,0 +1,413 @@
+
+### Types
+
+The following code example adds a new `MyEnum` type definition to Strapi v3:
+
+```jsx
+// path: ./api/foo/config/schema.graphql.js
+
+module.exports = {
+  definition: `
+    enum MyInput {
+      a
+      b
+      c
+    }
+  `,
+}
+```
+
+The Strapi v3 code example above should be replaced by the following code in Strapi v4:
+
+```jsx
+// path: ./src/index.js
+
+module.exports = {
+  register({ strapi }) {
+    const extensionService = strapi.plugin('graphql').service('extension');
+
+    extensionService.use(({ nexus }) => {
+      const MyEnum = nexus.enumType({
+        name: 'MyEnum',
+        members: ['a', 'b', 'c'],
+      });
+
+      return { types: [MyEnum] };
+    });
+  }
+}
+```
+
+:::note
+It's recommended to use the [nexus](https://nexusjs.org/) definition instead of raw [SDL](https://graphql.org/learn/schema/), but it’s still possible to use `typeDefs` to write raw SDL.
+:::
+
+### Queries
+
+The following code example adds a new query definition to Strapi v3:
+
+```jsx
+// path: ./api/foo/config/schema.graphql.js
+
+module.exports = {
+  query: `
+    myQuery(id: ID, status: MyInput, limit: Int): [MyQuery]!
+  `,
+  resolver: {
+    Query: {
+      myQuery: {
+        resolver: 'application::api-name.content-type-name.customFind',
+        // OR
+        resolver: async (obj, options, { context }) => {
+          await strapi.controllers.content-type-name.customFind(context);
+
+          return context.body.myQuery || [];
+        }
+      }
+    }
+  },
+}
+```
+
+The Strapi v3 code example above should be replaced by the following code in Strapi v4:
+
+```jsx
+// path: ./src/index.js
+
+module.exports = {
+  register({ strapi }) {
+    const extensionService = strapi.plugin('graphql').service('extension');
+
+    extensionService.use(({ nexus }) => {
+      const myQuery = nexus.extendType({
+        type: 'Query',
+        definition(t) {
+          //  myQuery definition
+          t.field('myQuery', {
+            // Response type
+            type: nexus.nonNull(nexus.list('MyQuery')),
+
+            // Args definition
+            args: { id: 'ID', status: 'MyInput', limit: 'Int' },
+
+            // Resolver definition
+            resolve(parent, args, context) {
+              const { id, status, limit } = args;
+      
+              return strapi.service('api::api-name.content-type-name').customFind(id, status, limit);
+            }
+          });
+        }
+      });
+
+      return { types: [myQuery] };
+    });
+  }
+}
+```
+
+::: note NOTES
+
+* In Strapi v4, REST controllers and GraphQL resolvers are not coupled anymore. The business logic is implemented in services and called either from the controller or the resolver. This approach keeps the business logic in one place so both REST and GraphQL can be customized the way you want.
```

Comment:
Honestly now that I'm seeing this written down my quesiton becomes how do we properly control access in U&P if the logic differs between REST/GraphQL since technically they are still linked at the permission level.

CC @Convly 

(This is entirely unrelated to the migration guide but technically this statement isn't entirely true)